### PR TITLE
feat(db): Add DB creation and population scripts

### PR DESCRIPTION
**Pull request by jorge-sanz**

### CODE REVIEW

File: db/create-db.sql [hunk 1]
```diff
@@ -0,0 +1,116 @@
+DROP DATABASE IF EXISTS halp;
+CREATE DATABASE halp;
+
+\c halp;
+
+DROP TABLE IF EXISTS employed;
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'Employed' -> 'Employee'

File: db/create-db.sql [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+DROP TABLE IF EXISTS patient;
+DROP TABLE IF EXISTS book;
+DROP TABLE IF EXISTS plate;
+DROP TABLE IF EXISTS menu;
+DROP TABLE IF EXISTS item;
+DROP TABLE IF EXISTS patientOrder;
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'patientOrder' -> 'order'

File: db/create-db.sql [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+DROP TABLE IF EXISTS copyBook;
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'copyBook' -> 'bookCopy'

File: db/create-db.sql [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+DROP TABLE IF EXISTS copyItem;
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'copyItem' -> 'itemCopy'

File: db/create-db.sql [hunk 5]
```diff
@@ -0,0 +1,116 @@
+DROP DATABASE IF EXISTS halp;
+CREATE DATABASE halp;
+
+\c halp;
+
+DROP TABLE IF EXISTS employed;
+DROP TABLE IF EXISTS patient;
+DROP TABLE IF EXISTS book;
+DROP TABLE IF EXISTS plate;
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'Plate' -> 'Dish'

File: db/create-db.sql [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+DROP TABLE IF EXISTS menu;
+DROP TABLE IF EXISTS item;
+DROP TABLE IF EXISTS patientOrder;
+DROP TABLE IF EXISTS copyBook;
+DROP TABLE IF EXISTS copyItem;
+
```
Comment (jorge-sanz) [inline db/create-db.sql]: Missing: 'purchase', 'bookLoan', 'menuChoice' and 'menuDish' tables

File: db/create-db.sql [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+CREATE TYPE employedType AS ENUM ('book', 'kiosk', 'food', 'assistant');
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'employedType' -> role

Missing 'manager' role.

File: db/create-db.sql [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+CREATE TYPE dishType AS ENUM ('starter', 'mainDish', 'dessert');
```
Comment (jorge-sanz) [inline db/create-db.sql]: Adding this Enum to Diagram in Astah.

Comment (jorge-sanz) [inline db/create-db.sql]: 'starter' -> 'first'
'mainDish' -> 'second'

File: db/create-db.sql [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+CREATE TYPE menuType AS ENUM ('breakfast', 'lunch', 'dinner');
```
Comment (jorge-sanz) [inline db/create-db.sql]: Finally, we will remove this enum.

Just one attribute for `Menu`: date, with day and time as a primary key.

File: db/create-db.sql [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+CREATE TABLE employed (
+  userName VARCHAR PRIMARY KEY,
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'userName' -> 'username'. It's one word.

File: db/create-db.sql [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+DROP TABLE IF EXISTS patientOrder;
+DROP TABLE IF EXISTS copyBook;
+DROP TABLE IF EXISTS copyItem;
+
+
+CREATE TYPE employedType AS ENUM ('book', 'kiosk', 'food', 'assistant');
+CREATE TYPE dishType AS ENUM ('starter', 'mainDish', 'dessert');
+CREATE TYPE menuType AS ENUM ('breakfast', 'lunch', 'dinner');
+
+CREATE TABLE employed (
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'employed' -> 'employee'

File: db/create-db.sql [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+  userName VARCHAR PRIMARY KEY,
+  password VARCHAR NOT NULL,
+  name VARCHAR NOT NULL,
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'name', 'lastname' and 'email' are not in the diagram. They should be removed.

File: db/create-db.sql [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+  lastname VARCHAR NOT NULL,
+  emal VARCHAR NOT NULL,
+  type employedType NOT NULL
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'type' -> 'role'

'employedType' -> 'role'

File: db/create-db.sql [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+);
+
+
+CREATE TABLE patient (
+  userName VARCHAR PRIMARY KEY,
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'userName' -> 'username'

File: db/create-db.sql [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+  password VARCHAR NOT NULL,
+  name VARCHAR NOT NULL,
+  lastname VARCHAR NOT NULL,
+  description VARCHAR
+);
```
Comment (jorge-sanz) [inline db/create-db.sql]: Missing attributes:
- birthday : DATE
- room : INTEGER

File: db/create-db.sql [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+CREATE TABLE book (
+  ID SERIAL PRIMARY KEY,
+  name VARCHAR NOT NULL,
+  author VARCHAR,
+  description VARCHAR NOT NULL,
+  image VARCHAR NOT NULL
+);
+
+
+CREATE TABLE plate (
```
Comment (jorge-sanz) [inline db/create-db.sql]: 'plate' -> 'dish'

File: db/create-db.sql [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+CREATE TABLE item (
+  ID SERIAL PRIMARY KEY,
+  name VARCHAR NOT NULL,
+  description VARCHAR NOT NULL,
+  image VARCHAR NOT NULL,
+  price INTEGER NOT NULL
+);
+
+
+CREATE TABLE patientOrder (
```
Comment (jorge-sanz) [inline db/create-db.sql:73]: 'patientOrder' -> 'order'

File: db/create-db.sql [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+  ID INTEGER PRIMARY KEY,
+  orderDate TIMESTAMP NOT NULL,
+  idPatient VARCHAR REFERENCES patient(userName)
+);
+
+
+CREATE TABLE copyBook (
+  referenceNumber VARCHAR PRIMARY KEY,
+  idBook INTEGER REFERENCES book(ID)
```
Comment (jorge-sanz) [inline db/create-db.sql]: Missing attribute: 'reserved: BOOLEAN'

File: db/create-db.sql [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+
+
+CREATE TABLE menuChoice (
+  ID INTEGER PRIMARY KEY REFERENCES patientOrder(ID),
+  idStarter INTEGER REFERENCES plate(ID) NOT NULL,
+  idMainDish INTEGER REFERENCES plate(ID) NOT NULL,
+  idDessert INTEGER REFERENCES plate(ID) NOT NULL
+);
+
+CREATE TABLE dishInMenu (
```
Comment (jorge-sanz) [inline db/create-db.sql]: Missing attribute: 'type : SMALLINT' with a new table for the enum.

**Review by jorge-sanz — COMMENTED** [review]
Some changes to have code and diagram synced.

File: db/create-db.sql (hunk 2 above)
Comment (josedl4) [reply db/create-db.sql]: order is a reserved word in sql

**Pull request merged**